### PR TITLE
Run on the device instead of using UVM to run on the host

### PR DESCRIPTION
**Pull request by Rombur**
Currently the code works by running on the host using UVM.

### CODE REVIEW

File: packages/Discretization/src/DTK_PointSearch_def.hpp [hunk 1]
```diff
@@ -307,39 +307,62 @@ std::tuple<Kokkos::View<int *, DeviceType>, Kokkos::View<int *, DeviceType>,
...
+        "flatten_query_ids", n_ref_pts );
+    Kokkos::View<Point *, DeviceType> flatten_ref_pts( "flatten_ref_pts",
+                                                       n_ref_pts );
+    unsigned int n_copied_pts = 0;
     for ( unsigned int topo_id = 0; topo_id < DTK_N_TOPO; ++topo_id )
     {
         unsigned int const size = _query_ids[topo_id].extent( 0 );
+
+        // First fill cell_indices. This has to be done on the host because
+        // _cell_indices_map only exists on the host.
```
Comment (dalg24) [inline packages/Discretization/src/DTK_PointSearch_def.hpp]: Good catch. How do you realize it?

File: packages/Discretization/src/DTK_PointSearch_def.hpp [hunk 2]
```diff
@@ -307,39 +307,62 @@ std::tuple<Kokkos::View<int *, DeviceType>, Kokkos::View<int *, DeviceType>,
...
+        auto cell_indices_per_topo_host =
```
Comment (dalg24) [inline packages/Discretization/src/DTK_PointSearch_def.hpp]: `topo_cell_indices_host`

File: packages/Discretization/src/DTK_PointSearch_def.hpp [hunk 3]
```diff
@@ -307,39 +307,62 @@ std::tuple<Kokkos::View<int *, DeviceType>, Kokkos::View<int *, DeviceType>,
...
-            query_ids_host( pos ) = _query_ids[topo_id]( i );
-            for ( unsigned int d = 0; d < _dim; ++d )
-                ref_pts_host( pos )[d] = _reference_points[topo_id]( i, d );
-            ++pos;
+            flatten_cell_indices_host( i + n_copied_pts ) =
+                _cell_indices_map[topo_id][cell_indices_per_topo_host( i )];
         }
+
+        // Fill flatten_query_ids
+        auto query_ids = _query_ids[topo_id];
```
Comment (dalg24) [inline packages/Discretization/src/DTK_PointSearch_def.hpp]: `topo_query_ids`

File: packages/Discretization/src/DTK_PointSearch_def.hpp [hunk 4]
```diff
@@ -307,39 +307,62 @@ std::tuple<Kokkos::View<int *, DeviceType>, Kokkos::View<int *, DeviceType>,
...
+                              Kokkos::RangePolicy<ExecutionSpace>( 0, size ),
+                              KOKKOS_LAMBDA( int const i ) {
+                                  flatten_query_ids( i + n_copied_pts ) =
+                                      query_ids( i );
+                              } );
+        Kokkos::fence();
+
+        // Fill flatten_ref_pts
+        unsigned int dim = _dim;
+        auto ref_pts = _reference_points[topo_id];
```
Comment (dalg24) [inline packages/Discretization/src/DTK_PointSearch_def.hpp]: `topo_`

File: packages/Discretization/src/DTK_PointSearch_def.hpp (hunk 1 above)
Comment (Rombur) [reply packages/Discretization/src/DTK_PointSearch_def.hpp]: I was thinking that I had a reason not to use `parallel_for` when I wrote this the first time. So I checked for data that only leaved on the host 🙂

**Comment by Rombur** [comment]
I have changed the name of variables, i.e., removed `flatten_` and add `topo_` as prefix.

File: packages/Discretization/src/DTK_PointSearch_def.hpp [hunk 5]
```diff
@@ -307,39 +307,59 @@ std::tuple<Kokkos::View<int *, DeviceType>, Kokkos::View<int *, DeviceType>,
...
     for ( unsigned int topo_id = 0; topo_id < DTK_N_TOPO; ++topo_id )
     {
         unsigned int const size = _query_ids[topo_id].extent( 0 );
+
+        // First fill cell_indices. This has to be done on the host because
+        // _cell_indices_map only exists on the host.
+        auto topo_cell_indices_host =
+            Kokkos::create_mirror_view( _cell_indices[topo_id] );
+        Kokkos::deep_copy( topo_cell_indices_host, _cell_indices[topo_id] );
         for ( unsigned int i = 0; i < size; ++i )
```
Comment (dalg24) [inline packages/Discretization/src/DTK_PointSearch_def.hpp]: Why is this one not a `parallel_for`?

**Pull request merged**